src/OrleansProviders/Streams/Common/PooledCache/FixedSizeBuffer.cs

```diff
@@ -69,17 +75,24 @@ public bool TryGetSegment(int size, out ArraySegment<byte> value)
             }
             value = new ArraySegment<byte>(buffer, count, size);
             count += size;
+            this.SegmentCount++;
             return true;
         }
 
         /// <summary>
-        /// Reset state and calls purge with its buffer, indicating that any segments using that address are no longer valid.
+        /// Trigger purge action on purge cancdidate buffer
         /// </summary>
-        public override void SignalPurge()
+        public override void SignalPurge(IDisposable purgeCandidate)
         {
-            count = 0;
-            purgeAction?.Invoke(this);
-            purgeAction = null;
+            purgeAction?.Invoke(purgeCandidate);
```

Comment:
please keep purgeAction = null for safety.
Under no circumstances should purge action ever be called more than once.